rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1713,6 +1720,35 @@ public JobIdData reSubmit(String sessionId, String jobId, PathSegment pathSegmen
         return mapper.map(newJobId, JobIdData.class);
     }
 
+    @Override
+    public WorkflowDescription getDescription(String sessionId, String jobId) throws IOException, RestException {
+        Scheduler scheduler = checkAccess(sessionId, "/scheduler/jobs/" + jobId + "/variables");
+        SchedulerSpaceInterface space = getSpaceInterface(sessionId);
+
+        try {
+            String jobXml = scheduler.getJobContent(JobIdImpl.makeJobId(jobId));
+            Job job;
+            try (InputStream tmpWorkflowStream = IOUtils.toInputStream(jobXml, Charset.forName(FILE_ENCODING))) {
+                job = JobFactory.getFactory().createJob(tmpWorkflowStream, null, null, scheduler, space);
+            }
+            WorkflowDescription workflowDescription = new WorkflowDescription();
+            workflowDescription.setName(job.getName());
+            workflowDescription.setProjectName(job.getProjectName());
+            workflowDescription.setDescription(job.getDescription());
+            workflowDescription.setVariables(job.getVariables());
```

Comment:
What if the job has no variables (job.getVariables() is null) ?

Reply:
variables is never null

generic info can be null